M2/cmake/check-libraries.cmake

```diff
@@ -352,6 +352,13 @@ else()
   unset(FACTORY_STREAMIO CACHE)
 endif()
 
+if(FLINT_FOUND)
+  check_include_files(flint/nmod.h HAVE_FLINT_NMOD_H)
+else()
+  # if flint isn't found, then we'll build a new version that has nmod.h
+  set(HAVE_FLINT_NMOD_H 1)
```

Comment:
Don't do this, it'll mess things up if the user wants to switch from building with Flint 3 to another version or vice versa. Just use `unset(HAVE_FLINT_NMOD_H CACHE)`, and if Flint is being built, then the variable will be set after that's done.

Reply:
Ok, cool.  I wasn't sure if what I was doing was the best approach.  :)  I just force-pushed a new version of this commit.